### PR TITLE
Add Head.compactable method

### PR DESCRIPTION
**Pull request by zhulongcheng**
Extract whether head has a compactable range  into a method (`Head.compactable` ).

### CODE REVIEW

File: head.go [hunk 1]
```diff
@@ -1020,6 +1020,13 @@ func (h *Head) MaxTime() int64 {
 	return atomic.LoadInt64(&h.maxTime)
 }
 
+// compactable returns whether the head has a compactable range.
+func (h *Head) compactable() bool {
+	// The head has a compactable range if 1.5 level 0 ranges are between the oldest
+	// and newest timestamp. The 0.5 acts as a buffer of the appendable window.
```
Comment (krasi-georgiev) [inline head.go]: you can combine the func comment with this one to make it only a single comment describing the func

Comment (zhulongcheng) [reply head.go]: ok

File: head.go [hunk 2]
```diff
@@ -1020,6 +1020,13 @@ func (h *Head) MaxTime() int64 {
 	return atomic.LoadInt64(&h.maxTime)
 }
 
+// compactable returns whether the head has a compactable range.
+// The head has a compactable range if 1.5 level 0 ranges are between the oldest
+// and newest timestamp.
```
Comment (krasi-georgiev) [inline head.go]: how about something like 
```
// compactable returns whether the head has a compactable range.
// The head has a compactable range when the head time range is 1.5 times the chunk range.
// The 0.5 acts as a buffer of the appendable window.
```

File: db.go [hunk 3]
```diff
@@ -409,13 +409,11 @@ func (db *DB) compact() (err error) {
 			return nil
 		default:
 		}
-		// The head has a compactable range if 1.5 level 0 ranges are between the oldest
-		// and newest timestamp. The 0.5 acts as a buffer of the appendable window.
-		if db.head.MaxTime()-db.head.MinTime() <= db.opts.BlockRanges[0]/2*3 {
```
Comment (krasi-georgiev) [inline db.go]: wondering why it was `<=` here and not just `<`

@gouthamve 	do you think this will cause issues?

File: head.go (hunk 2 above)
Comment (zhulongcheng) [reply head.go]: Great 👍 

Thanks.

**Comment by krasi-georgiev** [comment]
LGTM

ping @codesome

**Comment by zhulongcheng** [comment]
ping

**Pull request merged**